Performing Data Wrangling and EDA (Exploratory Data Analysis)

YolaKamalita opened the pull request:
Resolving #3

Collaborators
@YolaKamalita 
@M-MWW 
@czh000101 

What's remaining?
- Adding description to the codes, graphs, and tables.

What's done?

- Remove notebooks dir, and replace it with quarto dir.
- Move data_processing.Rmd to quarto dir.
- Add scatterplot.R (EDA) to quarto dir.
- Add Group_10_Analysis.qmd to combine data_processing.Rmd and scatterplot.R (EDA), and also add boxplot.

YolaKamalita committed:
move the data processing.Rmd files

YolaKamalita committed:
remove submodule notebooks

YolaKamalita committed:
remove notebooks dir, create quarto dir, move data_processing.Rmd to quarto dir

YolaKamalita committed:
combine data preprocessing and eda

YolaKamalita committed:
add scatterplot.R

YolaKamalita commented on quarto/Group_10_Analysis.qmd:
````diff
@@ -0,0 +1,188 @@
+
+## Missing value
+
+```{r}
````
From @M-MWW: Calculate the number of NA, if the proportion is not large, we can remove it, if the number is large, maybe we need to fill with the mean value.

M-MWW replied:
yes,calculate the number of NA and visualize it

YolaKamalita commented on quarto/Group_10_Analysis.qmd:
````diff
@@ -0,0 +1,188 @@
+text(x = 1:length(na_sum)+0.2, y = na_sum + 2, labels = na_sum, pos = 3, cex = 0.8, col = "red")
+```
+
+```{r}
````
From @M-MWW: for each type of movie, select them individually, and then calculate their average rating? then we can know what kind of movies have higher rating.

M-MWW replied:
See how many unique values in the categorical variables

YolaKamalita committed:
update comment from Readme

M-MWW committed:
add a data analysis

for each type of movie, select them individually, and then calculate their average rating， then we can know what kind of movies have higher rating

M-MWW committed:
report a error

M-MWW committed:
add a correlation coefficient

correlation coefficient between all the variables in df

M-MWW commented on quarto/Group_10_Analysis.qmd:
```diff
@@ -0,0 +1,214 @@
+  geom_bar(position="dodge", stat="count") +
+  labs(y = "Proportion")
+
+grid.arrange(p1, p2, p3, p4, ncol=2)#This part of the code has reported an error, but I can not solve it, I do not know what the problem is
```
This part of the code has reported an error, but I can not solve it, I do not know what the problem is

YolaKamalita committed:
change output format to be pdf

YolaKamalita committed:
deprecate logic to replace NA with mode, improving the graphs

chelsea7711 commented on quarto/Group_10_Analysis.qmd:
```diff
@@ -0,0 +1,281 @@
+# Summary statistics with adjusted skim()
+my_skim <- skim_with(base = sfl(n = length), 
+                     numeric = sfl(p0 = NULL, p100 = NULL,hist = NULL))
+knit_print(my_skim(IMDB ))
```
Adding genre as a group variable to skim()

chelsea7711 commented on quarto/Group_10_Analysis.qmd:
```diff
@@ -0,0 +1,281 @@
+  labs(y = "Votes", x = "Rating > 7")
+
+# Rating > 7 vs. Genre
+p6 <- ggplot(df, aes(x= genre,  y = ..prop.., group=rating_higher_than_7, fill=rating_higher_than_7)) +
```
1. Adjusting the codes (Incomplete pdf output)
2. Adding the legend for barplot

chelsea7711 commented on quarto/Group_10_Analysis.qmd:
```diff
@@ -0,0 +1,281 @@
+#| echo: true
+#| warning: false
+
+# Creating scatterplot between rating and explanatory variables
```
Adding regression line to the scatterplot.

YolaKamalita committed:
addressing feedbacks, adding description

YolaKamalita committed:
fix typo

Xiaoyu0077 commented on quarto/Group_10_Analysis.qmd:
```diff
@@ -0,0 +1,281 @@
+
+# Create new columns: rating_higher_than_7
+df <- df %>% 
+  mutate(rating_higher_than_7 = ifelse(rating <= 7, 0, 1))
```
Set a new column named rating_higher_than_7. If rating is higher than 7 ,the value will be 1, or it will be 0.
